driver-sync/src/main/com/mongodb/client/TransactionBody.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+/**
+ * A functional interface representing the body of a transaction.  Implementations of this interface should be idempotent in order to
+ * support transaction retry logic.
+ *
+ * @param <T> the return type
+ * @see ClientSession#withTransaction(TransactionBody)
+ * @since 3.10
+ */
+public interface TransactionBody<T> {
```

Comment:
Not sure about the name for this.  Didn't see another functional interface to use in the JDK.  The closest seems to be java.util.function.Supplier, but calling the method get() is weird in this context.

Reply:
`Transaction` reads to me like something that would actually start and commit the transaction.  But `TransactionBody` is awkward.  I'll look around for inspiration in other similar libraries.